lizard_ext/lizardfans.py

```diff
@@ -27,28 +32,35 @@ def __call__(tokens, reader):
         So don't store any data with an extension object.
         """
         reader.context.fileinfo.fan_in = reader.context.fileinfo.fan_out = 0
-        for token in tokens:
-            yield token
+        return tokens
 
     def fans(self, fileinfo):
         """
         Preparation of calculating the fan in and fan out (prototype)
         """
+        infile = Infile()
         self.name_list = [func.name for func in fileinfo.function_list]
         for func in fileinfo.function_list:
-            body = self.method_open(func)
+            if not func.filename == infile.name:
+                infile = self.open_method(func.filename)
+            body = self.store_method(infile, func)
             self.calculate_fan_in_fan_out(self.name_list, body, fileinfo, func)
```

Comment:
The name "calculate_fan_in_fan_out" is "lying." It doesn't just calculate, it also updates the state. So, I'm suggesting either:
1. rename it, or
2. Split the calculating and updating into two methods, to be come something like: self.update_fans(self.calcuate_fans(...), ...)

If you do the 2nd one, perhaps you will see a new potential class emerges.